src/magma/vk/ShaderModule.cpp

```diff
@@ -0,0 +1,63 @@
+#include "magma/vk/ShaderModule.h"
+
+#include <fstream>
+
+#include "magma/app/log.hpp"
+#include "magma/vk/vulkan_common.h"
+
+static std::vector<char> readFile(const std::string &filename)
+{
+    std::ifstream file(filename, std::ios::ate | std::ios::binary);
+
+    if (!file.is_open()) {
+        LOG_AND_THROW std::runtime_error("Failed to open file!");
+    }
+
+    size_t fileSize = (size_t) file.tellg();
+    std::vector<char> buffer(fileSize);
+    file.seekg(0);
+    file.read(buffer.data(), fileSize);
+
+    file.close();
+
+    return buffer;
+}
+
+static vk::ShaderStageFlagBits stageToVkStage(ShaderStage stage) {
```

Comment:
this function was static method of Shader class.
Also this method should be public